base/daemon/BaseDaemon.cpp

```diff
@@ -279,18 +271,27 @@ class SignalListener : public Poco::Runnable
     void onFault(
         int sig,
         const siginfo_t & info,
-        const ucontext_t & context,
+        const mcontext_t & mcontext,
         const StackTrace & stack_trace,
         UInt32 thread_num,
-        const std::string & query_id,
         DB::ThreadStatus * thread_ptr) const
     {
         DB::ThreadStatus thread_status;
 
+        String query_id;
+        String query;
+
         /// Send logs from this thread to client if possible.
         /// It will allow client to see failure messages directly.
         if (thread_ptr)
         {
+            query_id = thread_ptr->getQueryId().toString();
```

Comment:
Each extra memory read may lead to SIGSEGV from the SIGSEGV handler, but I guess this is fine to obtain query_id/query from the thread_ptr.